Pull upstream and support initial pose.

RobertMilijas opened the pull request.

MichaelGrupp committed:
Fix segfault by changing the destruction order. (#1235)

The metrics registry is used as a raw pointer reference in map builder
and must outlive it.

Since C++ destroys in reverse declaration order, we have to put the
registry declaration before the map builder bridge.

Fixes #1234.

MichaelGrupp committed:
Replace a few string operator+ by absl::StrCat(). (#1244)

...in some places that can be called frequently.

MichaelGrupp committed:
Extend trajectory export tool to write TransformStamped topics. (#1169)

See discussion in #1166.

fprott committed:
Landmarks note (#1242)

Added a note to point out that even if all the SLAM options are set for a 2D SLAM the landmarks will remain 3D objects.

fprott committed:
Expanded Landmarks documentation (#1241)

Expanded landmarks to included more information and make it more clear and first-time user friendly. Used knowledge from the issues, especially issue https://github.com/googlecartographer/cartographer_ros/issues/909 and https://github.com/googlecartographer/cartographer_ros/issues/1067

MichaelGrupp committed:
Update clang-format to new Google style. (#1260)

Apparently the format bot uses a bleeding edge clang-format that uses
the new Google style and reformats a bunch of files in every PR. This is
an empty commit to trigger this in a separate commit.

See https://github.com/llvm-mirror/clang/commit/62e3198c4f5490a1c60ba51d81fe2e1f0dc99135

MichaelGrupp committed:
Unify trajectory state checks for service handlers. (#1262)

Some service handlers need to check if a trajectory is in a valid
state for the requested operation. If it's not, they return an error
response.

`Node::CheckTrajectoryState` allows to avoid code duplication in such
situations.

ashnarayan13 committed:
Updated doc to remove abseil-cpp from ROS (#1211)

I have added a few lines to the doc to alert users to remove ros abseil-cpp and also to change the rosinstall file if they wish to checkout a different version of cartographer. As discussed in #1208

MichaelGrupp committed:
Add explicit --save_state_filename to offline node options. (#1204)

This allows to serialize the state also when no bagfile is given, e.g.
when the offline node is used to run an optimization and/or trimming
configuration on a pbstream. Or simply when one wants to use a custom
name directly. This doesn't break compatibility with the previous CLI.

MichaelGrupp committed:
Service to query trajectory segments from the pose graph. (#1222)

This `/trajectory_query` service allows to look up trajectory segments
from the pose graph.

This can be useful if one wants to get optimized trajectory data from
the current SLAM run and listening to live TF data would be too noisy.
For example, to stitch buffered point cloud data from a depth sensor
based on a recent localization trajectory segment of a robot.

Before, the pose graph trajectory nodes were not available except for
the trajectory marker topic (only positions, no orientation; inefficient)
or after serialization (which is not practical in live operation).

mgladkova committed:
Simplify start_trajectory service (RFC-28) (#1245)

* Simplify start_trajectory.

* Ran clang-format.

* Make corrections based on the review

* Make corrections based on review #2, remove obsolete functions

* Remove unnecessary local variables

mgladkova committed:
Update ROS API documentation with description of changed/new services (#1282)

tfoote committed:
Remove EOL Distros from CI matrix (#1379)

Indigo and Lunar are both EOL

MichaelGrupp committed:
Remove unnecessary eigen_conversions dependency. (#1278)

Seems to have been used in the past, but isn't needed anymore.

RobertMilijas committed:
Merge remote-tracking branch 'upstream/master' into erl

Conflicts:
	cartographer_ros/cartographer_ros/map_builder_bridge.cc
	cartographer_ros/cartographer_ros/node.cc
	cartographer_ros/cartographer_ros/node.h
	cartographer_ros/cartographer_ros/trajectory_options.cc
	cartographer_ros_msgs/msg/TrajectoryOptions.msg

fpetric commented:
@ojura can you just glance over this quickly to confirm it won't break our existing setups?

RobertMilijas committed:
Fix bug

RobertMilijas committed:
Support for initial pose of first trajectory.

Allow the user to specify a trajectories relative pose with respect to the
origin by passing -1 as the trajectory_id.

RobertMilijas committed:
Add comment

ojura committed:
Code review of merging 1de03b3

RobertMilijas committed:
Merge pull request #1 from RobertMilijas/erl-devel

Support for initial pose of first trajectory.

RobertMilijas commented:
Thnx for the review, @ojura.
